lib/features/family/features/reflect/presentation/pages/grateful_screen.dart

```diff
@@ -62,12 +67,19 @@ class _GratefulScreenState extends State<GratefulScreen> {
             children: [
               GratefulAvatarBar(
                 uiModel: uiModel.avatarBarUIModel,
-                onAvatarTapped: _cubit.onAvatarTapped,
+                onAvatarTapped: (int i) async {
```

Comment:
Personally I would revert this change. Then below in handleCustom, the data gives you the profile. And then there I would do the setActiveProfile.

Reply:
Good idea!